source/extensions/matching/http/metadata_input/meta_input.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "envoy/extensions/matching/common_inputs/network/v3/network_inputs.pb.h"
+#include "envoy/extensions/matching/common_inputs/network/v3/network_inputs.pb.validate.h"
+#include "envoy/http/filter.h"
+#include "envoy/matcher/matcher.h"
+
+#include "source/common/config/metadata.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Matching {
+namespace Http {
+namespace MetadataInput {
+
+class MetadataMatchData : public ::Envoy::Matcher::CustomMatchData {
+public:
+  explicit MetadataMatchData(const ProtobufWkt::Value& value) : value_(value) {}
+  const ProtobufWkt::Value& value_;
```

Comment:
note to myself: this is a dangerous capture by reference. I see CEL does the same thing, but we probably need to audit for potential dangling references.